encodings/fastlanes/src/for/compute/compare.rs

```diff
@@ -53,10 +51,13 @@ where
         if let Some(reference) = reference {
             rhs = rhs.wrapping_sub(&reference);
         }
-        if lhs.shift() > 0 {
-            // Since compare requires that both sides are of same dtype this will always succeed and not panic
-            rhs = rhs >> (lhs.shift() as u32)
-        }
+        // Since compare requires that both sides are of same dtype this will always succeed and not panic
```

Comment:
This comment is wrong now? The issue is that the sub is wrapping so it will succeed and produce 0 instead of skipping the comparison. Would it be simpler to detect this case and short circuit?

Reply:
IE it only happens for values < reference?